erpnext/regional/report/vat_audit_report/vat_audit_report.py

```diff
@@ -0,0 +1,246 @@
+					continue
+
+	def get_item_amount_map(self, parent, item_code, taxes):
+		net_amount = abs(self.invoice_items.get(parent).get(item_code))
+		tax_rate = taxes[0]
+		tax_amount = taxes[1]
+		gross_amount = net_amount + tax_amount
+		item_amount_map = self.item_tax_rate.setdefault(parent, {}) \
+			.setdefault(item_code, [])
+		amount_dict = {
+			"tax_rate": tax_rate,
+			"gross_amount": gross_amount,
+			"tax_amount": tax_amount,
+			"net_amount": net_amount
+		}
+		item_amount_map.append(amount_dict)
+
+		return tax_rate, item_amount_map
+
+	def get_conditions(self):
+		conditions = ""
+		for opts in (("company", " and company=%(company)s"),
+			("from_date", " and posting_date>=%(from_date)s"),
+			("to_date", " and posting_date<=%(to_date)s")):
+			if self.filters.get(opts[0]):
+				conditions += opts[1]
+
+		return conditions
+
+	def get_data(self,doctype):
+		consolidated_data = self.get_consolidated_data()
+		section_name = _("Purchases") if doctype == "Purchase Invoice" else _("Sales")
+
+		for rate, section in consolidated_data.items():
+			rate = int(rate)
+			label = frappe.bold(_("Standard Rate") + " " + section_name + " " + str(rate) + "%")
+			section_head = {"posting_date": label}
+			total_gross = total_tax = total_net = 0
+			self.data.append(section_head)
+			for row in section.get("data"):
+				self.data.append(row)
+				total_gross += row["gross_amount"]
+				total_tax += row["tax_amount"]
+				total_net += row["net_amount"]
+
+			total = {
+				"posting_date": frappe.bold(_("Total")),
+				"gross_amount": total_gross,
+				"tax_amount": total_tax,
+				"net_amount": total_net,
+				"bold":1
+			}
+			self.data.append(total)
+			self.data.append({})
+
+	def get_consolidated_data(self):
+		consolidated_data_map={}
+		for inv, inv_data in self.invoices.items():
+			for rate, items in self.items_based_on_tax_rate.get(inv).items():
+				consolidated_data_map.setdefault(rate, {"data": []})
+				for item in items:
+					row = {}
+					item_details = self.item_tax_rate.get(inv).get(item)
+					row["account"] = inv_data.get("account")
+					row["posting_date"] = formatdate(inv_data.get("posting_date"), 'dd-mm-yyyy')
+					row["invoice_number"] = inv
+					row["party"] = inv_data.get("party")
+					row["remarks"] = inv_data.get("remarks")
+					row["gross_amount"]= item_details[0].get("gross_amount")
+					row["tax_amount"]= item_details[0].get("tax_amount")
+					row["net_amount"]= item_details[0].get("net_amount")
+					consolidated_data_map[rate]["data"].append(row)
+
+		return consolidated_data_map
+
+	def get_columns(self,doctype):
+		self.columns = [
+			{
+				"fieldname": "posting_date",
+				"label": "Posting Date",
+				"fieldtype": "Data",
+				"width": 200
+			},
+			{
+				"fieldname": "account",
+				"label": "Account",
+				"fieldtype": "Link",
+				"options": "Account",
+				"width": 140
+			},
+			{
+				"fieldname": "invoice_number",
+				"label": "Reference",
+				"fieldtype": "Link",
+				"options": doctype,
```

Comment:
Passing Doctype won't work, all the columns even Purchase Invoice refers to Sales Invoice
Add a column for for document type and make invoice number a Dynamic Link
You can refer accounts receivable report for reference

Reply:
Fixed